cosmic-applet-audio/src/pulse.rs

```diff
@@ -15,71 +15,80 @@ use libpulse_binding::{
     struct Connect;
 
     subscription::unfold(
         std::any::TypeId::of::<Connect>(),
         State::Init,
-        |state| async move {
-            match state {
-                State::Init => {
-                    let PulseHandle {
-                        to_pulse,
-                        from_pulse,
-                    } = PulseHandle::new();
-                    (
-                        Some(Event::Init(Connection(to_pulse))),
-                        State::Connecting(from_pulse),
-                    )
-                }
-                // Waiting for Connection to succeed
-                // The GUI doesn't have to monitor this state, as it is never sent to the GUI
-                State::Connecting(mut from_pulse) => match from_pulse.recv().await {
-                    Some(Message::Connected) => {
-                        (Some(Event::Connected), State::Connected(from_pulse))
-                    }
-                    Some(Message::Disconnected) => {
-                        (Some(Event::Disconnected), State::Connecting(from_pulse))
-                    }
-                    Some(m) => {
-                        panic!("Unexpected message: {:?}", m);
-                    }
-                    None => {
-                        panic!("Pulse Sender dropped, something has gone wrong!");
-                    }
-                },
-                State::Connected(mut from_pulse) => {
-                    // This is where we match messages from the pulse server to pass to the gui
-                    match from_pulse.recv().await {
-                        Some(Message::SetSinks(sinks)) => (
-                            Some(Event::MessageReceived(Message::SetSinks(sinks))),
-                            State::Connected(from_pulse),
-                        ),
-                        Some(Message::SetSources(sources)) => (
-                            Some(Event::MessageReceived(Message::SetSources(sources))),
-                            State::Connected(from_pulse),
-                        ),
-                        Some(Message::SetDefaultSink(sink)) => (
-                            Some(Event::MessageReceived(Message::SetDefaultSink(sink))),
-                            State::Connected(from_pulse),
-                        ),
-                        Some(Message::SetDefaultSource(source)) => (
-                            Some(Event::MessageReceived(Message::SetDefaultSource(source))),
-                            State::Connected(from_pulse),
-                        ),
-                        Some(Message::Disconnected) => {
-                            (Some(Event::Disconnected), State::Connecting(from_pulse))
-                        }
-                        None => (Some(Event::Disconnected), State::Connecting(from_pulse)),
-                        _ => (None, State::Connected(from_pulse)),
-                    }
+        |mut state| async move {
+            loop {
```

Comment:
is putting this in a loop more performant than just using iced to poll?

Reply:
And/or what happens is the applet tries to close, does it hang on the loop and/or crash?

Reply:
ah, then maybe we want to switch to [channel](https://docs.rs/iced/latest/iced/subscription/fn.channel.html)?

Reply:
we can add that as a todo if you'd like

Reply:
Ya that sounds good